redbot/cogs/mod/events.py

```diff
@@ -19,20 +20,30 @@ class Events(MixinMeta):
 
     async def check_duplicates(self, message):
         guild = message.guild
-        author = message.author
+        author = message.author.id
+        channel = message.channel
 
-        guild_cache = self.cache.get(guild.id, None)
-        if guild_cache is None:
-            repeats = await self.config.guild(guild).delete_repeats()
+        repeats = await self.config.guild(guild).delete_repeats()
+        repeats_channels = await self.config.guild(guild).delete_repeats_channels.all()
```

Comment:
The old cache only made config calls when it had not already been filled with data for a particular guild. Yours (inefficiently) does so on every message that is checked. You need to only call from config **when needed**.